### DISCUSSION TITLE
Can BouncyCastlePQCProvider be directly used for TLS 1.3 with BCJSSE?

### DISCUSSION THREAD

**Discussion by rsmagdum** [bcgit/bc-java#1848] [Q&A] [answered]
https://www.keyfactor.com/blog/post-quantum-algorithm-update-in-bouncy-castle/
https://www.youtube.com/watch?v=EABttUCoTdY
After reading/watching above doc/workshop, my understanding is we can use PQC algorithms from BC PQC provider for key generation, signging operations, but it's still not integrated to use with TLS 1.3 (BCJSSE) hybrid key exchange - where a shared secret shall be derived by concatenating traditional + pqc kem secret.
Is my understanding correct?

**Comment by peterdettman** [comment, marked as answer]
Yes, it requires separate work on the TLS implementation (which is in progress) to enable the hybrid key exchanges.

**Reply by rsmagdum** [reply]
Thanks for confirmation @peterdettman
